templates/distribution/scripts/pre-apply.sh.tpl

```diff
@@ -95,4 +95,63 @@ deleteGatekeeper
 
 {{- end }} # end distributionModulesPolicyType
 
+{{- if index .reducers "distributionModulesTracingType" }}
+
+deleteTempo() {
+
+  $kustomizebin build $vendorPath/modules/tracing/katalog/minio-ha > delete-tracing-minio-ha.yaml
+  $kustomizebin build $vendorPath/modules/tracing/katalog/tempo-distributed > delete-tracing-tempo-distributed.yaml
+
+{{- if eq .spec.distribution.modules.monitoring.type "none" }}
+  if ! $kubectlbin get apiservice v1.monitoring.coreos.com; then
+    cat delete-tracing-minio-ha.yaml | $yqbin 'select(.apiVersion != "monitoring.coreos.com/v1")' > delete-tracing-minio-ha-filtered.yaml
+    cp delete-tracing-minio-ha-filtered.yaml delete-tracing-minio-ha.yaml
+    cat delete-tracing-tempo-distributed.yaml | $yqbin 'select(.apiVersion != "monitoring.coreos.com/v1")' > delete-tracing-tempo-distributed-filtered.yaml
```

Comment:
I think that since the process of deletion on the tracing module checks if the monitoring is present and filters out resources, it will just delete what is present in that moment. 

I think that it should be safe since deleting a CRD will also delete all the CR automatically